src/store/actions/AnimateMap.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  AnimateMapStageOptions,
+  ReplicatedUser,
+  ReplicatedVenue,
+} from "../reducers/AnimateMap";
+import { ReduxAction } from "../../types/redux";
+
+export enum AnimateMapActionTypes {
+  UPDATE_STAGE_OPTIONS = "AnimateMapActionTypes.UPDATE_STAGE_OPTIONS",
+  UPDATE_ZOOM = "AnimateMapActionTypes.UPDATE_ZOOM",
+  UPDATE_USERS = "AnimateMapActionTypes.UPDATE_USERS",
+  UPDATE_VENUES = "AnimateMapActionTypes.UPDATE_VENUES",
+}
+
+export type setAnimateMapStageOptionsAction = ReduxAction<
+  AnimateMapActionTypes.UPDATE_STAGE_OPTIONS,
+  {
+    options: AnimateMapStageOptions;
+  }
+>;
```

Comment:
This file seems to be based on our legacy/outdated/tech debt redux patterns. For all new work we want to be following Redux best practices and use Redux Toolkit, as per:

- https://redux.js.org/style-guide/style-guide#use-redux-toolkit-for-writing-redux-logic
- https://redux-toolkit.js.org/
  - https://redux-toolkit.js.org/usage/usage-with-typescript
  - https://redux-toolkit.js.org/tutorials/typescript

---

I though there should be an example of existing code I could show you in the codebase currently, but I couldn't find it, so I expect it may be buried away in one of my open WIP PR's (if I can find it, I will link it here so you can see a 'real world code implemented example' of this)

In essence, we basically want to be using `createSlice` for the main bulk of Redux state setup:

- https://redux-toolkit.js.org/usage/usage-with-typescript#createslice

Potentially falling back to using `createAction` or other things in extreme edgecases (though most of the time `createSlice` will handle that need for us anyway:

- https://redux-toolkit.js.org/usage/usage-with-typescript#createaction

Sometimes `createAsyncThunk` will also be useful:

- https://redux-toolkit.js.org/usage/usage-with-typescript#createasyncthunk

This pattern batches all of the actions/reducer/etc into a single place, as is best practice, so instead of including this in `store/actions/*`, etc; we should just make the file directly in `store/` directly.